Step-by-step-Tutorials/Declarative-Persistence.md

```diff
@@ -228,6 +228,13 @@ For example, a stack of elements may be externally represented as a `List<T>`, b
 
 In the case of our `Manager` class, the `_me` field is simply a cached value, something we don't even need to keep as a field in the first place, it can be created any time we need it, but since it's going to be a commonly used value, it's worth keeping it around in a transient field.
 
+## Automatic loading of state
+
+If a grain type has state, at activation time the state will be loaded from storage and then `OnActivateAsync` is called so you can be sure that the state is loaded when initializing your grain. This is the only case that Orleans calls `ReadStateAsync` automatically. If you want to write the state or read it in some other place, you should do it on your own.
+
```

Comment:
I'd add something about the fact that the grain is the owner of its state. Therefore in a normal flow the grain shouldn't call `ReadStateAsync`. If a call to `WriteStateAsync` fails, then a call to `ReadStateAsync` makes sense.